src/JustSaying/Messaging/Monitoring/LogOperation.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Monitoring
+{
+    internal sealed class LogOperation : IDisposable
+    {
+        private readonly ILogger _logger;
+        private readonly string _message;
+        private readonly object[] _args;
+        private readonly Stopwatch _watch;
+
+        public LogOperation(ILogger logger, string message, params object[] args)
+        {
+            _logger = logger;
+            _message = message;
+            _args = args;
+            _watch = new Stopwatch();
+
+            _watch.Start();
+        }
+
+        public void Dispose()
+        {
+            var args = _args.Concat(new object[] { _watch.ElapsedMilliseconds }).ToArray();
```

Comment:
I don't understand what this is doing.

Reply:
The number of args passed to the log message needs to match the number of holes in the message. When using the API like this `_logger.Time("Executing operation called {OperationName}", "LoadingMessages")`, the message that is produced would be `"Executing operation called LoadingMessages completed in 100ms"`, and since the elapsed ms is also a log property, we need to concat the user-provided args with the operation args.

It's almost the same as what this library is doing: https://github.com/nblumhardt/serilog-timings. 

I added it to try and work out why some tests were hanging, but it turned out that goaws was hanging - restarting it fixed the issue. It might be worth leaving in as it creates symmetry with the `Operation` type we have that we use for stats timings, but for logs instead.